protocols/kad/CHANGELOG.md

```diff
@@ -6,8 +6,11 @@
 - Remove deprecated public modules `handler`, `protocol` and `kbucket`.
   See [PR 3896].
 
+- Remove `TUserData` in `KademliaHandler`, `KademliaHandlerEvent`, `KademliaHandlerIn`, `OutboundSubstreamState` and replace with `QueryId`. See [PR 3959].
```

Comment:
This is an internal refactoring now that the handler is private, correct? Non-user facing changes don't need a changelog entry.

Reply:
@thomaseizinger No problem, I thought that you wanted to keep a trace of all the changes even internally but no problem I've removed that :)